### PR TITLE
Add generic OpenAI-compatible LLM provider

### PR DESCRIPTION
**Pull request by titusio**
## Summary

Adds a generic **OpenAI-Compatible** provider for AI Cleanup, letting Achew
connect to any endpoint that implements the OpenAI REST API — e.g. LiteLLM,
vLLM, or another self-hosted gateway — via a base URL and optional API key.

Every existing provider hardcodes its endpoint (or uses a vendor SDK), so
there was previously no way to use a self-hosted OpenAI-compatible gateway.
This adds a single configurable provider that covers all of them.

## What it does

- Discovers models from `GET {base_url}/models` (no vendor-specific filtering)
- Generates via streamed `POST {base_url}/chat/completions`
- API key is optional, so keyless local gateways work
- Normalizes the base URL (adds a scheme if missing, strips a trailing slash)
- Falls back to a request without `response_format` if the backend rejects it,
  since a generic gateway fronts models that don't all support JSON mode

## Implementation

- New `OpenAICompatibleService`, registered in the provider registry
- Adds a `base_url` field and an `openai_compatible` section to the config
- Wires up the model-listing and cancel-changes routes
- No frontend changes needed — the settings UI is data-driven from `setup_fields`

## Docs & tests

- Documented in `docs/getting-started/setup-llm-providers.md`
  (summary-table row + setup example)
- Adds `tests/llm_providers/test_openai_compatible_service.py` — 8 tests covering
  URL normalization, validation, model listing, the streaming happy path, the
  `response_format` fallback, and empty input
- Also gitignores the repo-root `config/` and `cache/` dirs that the documented
  docker-compose workflow creates at runtime

## Testing

- `ruff check` / `ruff format` clean; `pyright` (basic mode) reports 0 errors
- Full backend suite passes (185 passed, 2 pre-existing xfailed)
- Verified end-to-end against a live LiteLLM instance: model discovery,
  validation, and AI Cleanup all succeed

### CODE REVIEW

**Comment by titusio** [comment]
Note: I did use AI to help me write this PR, reviewed the changes on my own though.

**Comment by bgaeddert** [comment]
I tested this locally. Works great. Only thing I might suggest is a check box to skip ssl verification. Thanks!

File: backend/app/services/llm_providers/openai_compatible_service.py [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+                                                f"Processed {result['total_parsed']}/{total_chapters} chapters",
+                                            )
+
+                    # Stream completed successfully; no need to try further variants.
+                    break
+
+            self._notify_progress(100, "Processing AI response…")
+
+            try:
+                response_data = json.loads(content_received)
```
Comment (SirGibblets) [inline backend/app/services/llm_providers/openai_compatible_service.py]: Depending on the underlying provider and model, I occasionally see responses wrapped in markdown fences (e.g. ```` ```json ... ``` ````) despite instructions. I'd recommend stripping such fences prior to parsing here.

File: backend/tests/llm_providers/test_openai_compatible_service.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+
+@pytest.fixture
+def configured(monkeypatch):
+    """Install a saved config pointing at the mocked endpoint."""
+    app_cfg = C.AppConfig()
+    app_cfg.llm.openai_compatible.base_url = BASE_URL
+    app_cfg.llm.openai_compatible.api_key = "sk-test"
+    monkeypatch.setattr(C, "_app_config", app_cfg)
+    return app_cfg
```
Comment (SirGibblets) [inline backend/tests/llm_providers/test_openai_compatible_service.py]: Noticed an issue here after seeing a couple of tests fail locally.

I think there's an edge-case dependency on the real local config here: the tests that *don't* use this fixture (`test_base_url_normalization`'s empty-string case and `test_validate_config_missing_base_url`) still end up reading the real config from disk. When `base_url` is falsy, `_get_base_url()` falls through to `get_app_config()`, which loads the real file.

As far as I can tell this will not affect clean installs (e.g. CI), but those two tests end up failing after a user has set up the provider on their local instance. As we'll want to keep the real config isolated from the tests, I suggest pinning the config for the whole module and layering `configured` on top, e.g.:

```suggestion
@pytest.fixture(autouse=True)
def isolated_config(monkeypatch):
    """Use an isolated default config for tests."""
    app_cfg = C.AppConfig()
    monkeypatch.setattr(C, "_app_config", app_cfg)
    return app_cfg


@pytest.fixture
def configured(isolated_config):
    """Point the isolated config at the mocked endpoint."""
    isolated_config.llm.openai_compatible.base_url = BASE_URL
    isolated_config.llm.openai_compatible.api_key = "sk-test"
    return isolated_config
```

**Review by SirGibblets — CHANGES_REQUESTED** [review]
Thanks for the contribution! Looks pretty solid so far, just a couple minor fixes needed.

**Comment by SirGibblets** [comment]
> Only thing I might suggest is a check box to skip ssl verification.

@bgaeddert An excellent suggestion, and one that could be helpful for other existing providers as well (Ollama/LM Studio). I'll open a separate issue for it to avoid expanding the scope of this PR.

File: backend/app/services/llm_providers/openai_compatible_service.py (hunk 1 above)
Comment (titusio) [reply backend/app/services/llm_providers/openai_compatible_service.py]: Good catch. Added a `_strip_code_fences` helper and run the accumulated content through it before `json.loads`, so responses wrapped in ```` ```json … ``` ```` now parse. Also added a regression test (`test_process_chapter_titles_strips_markdown_fences`). Fixed in a5be734.

File: backend/tests/llm_providers/test_openai_compatible_service.py (hunk 2 above)
Comment (titusio) [reply backend/tests/llm_providers/test_openai_compatible_service.py]: Applied your suggestion. Added an `autouse` `isolated_config` fixture and layered `configured` on top of it, so no test reads the real on-disk config anymore. Verified both previously-affected tests pass with a populated local config. Thanks! Fixed in a5be734.

**Review by SirGibblets — APPROVED** [review]
LGTM, thanks again for the contribution!
